### PR TITLE
Refactor util.misc

### LINKED ISSUE #761: Proposal: clean up napari.util.misc
I noticed that `napari.util.misc` is getting a bit bloated and I think it would be good to clean it up a bit. To get the conversation rolling, I propose the following:

- create an image utils file
    - `is_rgb()`
    - `is_pyramid()`
    - `trim_pyramid()`
    - `should_be_pyramid()`
    - `get_pyramid_and_rgb()`
    - `fast_pyramid()`
- create a qt utils file
    - `mouse_press_callbacks()`
    - `mouse_move_callbacks()`
    - `mouse_release_callbacks()`
- move `format_doc()` into `napari.util.naming` (the only place it is used)
- move `interpolate_coordinates()` to the labels layer model folder or file (the only place it is used)
- `compute_max_shape()` doesn't appear to be used anywhere. delete?
- `segment_normal_vector()` doesn't appear to be used anywhere. delete?

### PR DESCRIPTION
**Pull request by kevinyamauchi**
# Description
As discussed in #761, this PR moves the the util functions used for interactions (e.g., mouse event callbacks) from `/napari/util/misc.py` to `napari/util/interactions.py `. I also deleted two unused functions from `/napari/util.misc.py` and deleted some unused code that had been commented out.

## Type of change
- [x] Bug-fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update

# References
This PR closes #761!

# How has this been tested?
- [x]  Added tests for `ReadOnlyWrapper` in `/napari/util/tests/test_interactions.py`
- [x] Tests pass
- [x] Tested `/examples/add_image.py`

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: napari/util/misc.py [hunk 1]
```diff
@@ -214,38 +150,8 @@ def __str__(self):
         # render_kw_only_separator = True
         for param in self.parameters.values():
             formatted = str(param)
-
-            # kind = param.kind
-
-            # if kind == inspect._POSITIONAL_ONLY:
```
Comment (kevinyamauchi) [inline napari/util/misc.py:220]: @kne42 , I think you may have written `CallSignature()`. Is it okay to delete these commented out sections? I am happy to put them back in if we're saving them for later.

Comment (kne42) [reply napari/util/misc.py:220]: I put that there because it's a modified version of `inspect.Signature` but it should be ok to delete I think

Comment (kevinyamauchi) [reply napari/util/misc.py:220]: Got it. Thanks!

**Comment by kevinyamauchi** [comment]
@napari/core-devs, I think this is ready for review!

**Review by jni — APPROVED** [review]

**Comment by jni** [comment]
I'll let @kne42 or @sofroniewn push the button, but looks great to me! =)

**Review by sofroniewn — APPROVED** [review]
LGTM!

**Pull request merged**